packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2547,29 +2542,23 @@ export class ContainerRuntime
 		}
 	}
 
-	private async loadIdCompressor() {
-		if (
-			this._idCompressor === undefined &&
-			this.idCompressorMode !== undefined &&
-			this._loadIdCompressor === undefined
-		) {
-			this._loadIdCompressor = this.createIdCompressor()
-				.then((compressor) => {
-					// Finalize any ranges we received while the compressor was turned off.
-					const ops = this.pendingIdCompressorOps;
-					this.pendingIdCompressorOps = [];
-					for (const range of ops) {
-						compressor.finalizeCreationRange(range);
-					}
-					assert(this.pendingIdCompressorOps.length === 0, 0x976 /* No new ops added */);
-					this._idCompressor = compressor;
-				})
-				.catch((error) => {
-					this.logger.sendErrorEvent({ eventName: "IdCompressorDelayedLoad" }, error);
-					throw error;
-				});
+	private initIdCompressor() {
+		if (this._idCompressor === undefined && this.idCompressorMode !== undefined) {
+			try {
+				this._idCompressor = this.createIdCompressorFn();
+
+				// Finalize any ranges we received while the compressor was turned off.
+				const ops = this.pendingIdCompressorOps;
+				this.pendingIdCompressorOps = [];
+				for (const range of ops) {
+					this._idCompressor.finalizeCreationRange(range);
+				}
+				assert(this.pendingIdCompressorOps.length === 0, 0x976 /* No new ops added */);
+			} catch (error) {
+				this.logger.sendErrorEvent({ eventName: "IdCompressorDelayedLoad" }, error);
+				throw error;
+			}
 		}
-		return this._loadIdCompressor;
 	}
```

Comment:
The differences are:
1. Renamed 'loadIdCompressor' -> 'initIdCompressor'
2. The method is no longer async.
3. The method is no longer gated on 'this._loadIdCompressor' since we've removed the 'in-flight' state.
4. Promise '.then(..).catch(..)' replaced with normal try/catch.